NGitLab/Impl/MilestoneClient.cs

```diff
@@ -1,18 +1,31 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using NGitLab.Extensions;
 using NGitLab.Models;
 
 namespace NGitLab.Impl
 {
     public class MilestoneClient : IMilestoneClient
     {
+        internal enum Scope
+        {
+            Projects,
+            Groups,
+        }
+
         private readonly API _api;
         private readonly string _milestonePath;
 
-        public MilestoneClient(API api, int projectId)
+        internal MilestoneClient(API api, Scope scope, int id)
         {
             _api = api;
-            _milestonePath = $"{Project.Url}/{projectId.ToStringInvariant()}/milestones";
+            _milestonePath = $"/{scope.ToString().ToLowerInvariant()}/{id.ToStringInvariant()}/milestones";
+        }
+
+        [Obsolete("Use GitlabClient.GetMilestone() or GitlabClient.GetGroupMilestone() instead.")]
```

Comment:
Make sure to use correct PascalCase:
```suggestion
        [Obsolete("Use GitLabClient.GetMilestone() or GitLabClient.GetGroupMilestone() instead.")]
```